kafka/metrics.go

```diff
@@ -382,6 +394,15 @@ func (h *metricHooks) OnBrokerWrite(meta kgo.BrokerMetadata, _ int16, bytesWritt
 		int64(bytesWritten),
 		metric.WithAttributeSet(attribute.NewSet(attrs...)),
 	)
+
+	// Record latency when the request type is Produce.
+	if key == int16(kmsg.Produce) {
```

Comment:
This won't record the latency for failed writes, shouldn't those be recorded too?

Reply:
>This won't record the latency for failed writes

Could you elaborate a bit more? I don't think the key changes depending on the result of the write, but I may have missed something.

Reply:
If `err != nil` (on line 372) the code takes another branch above and returns immediately.

Reply:
Ah, I missed that, thanks!